src/uct/ib/base/ib_md.c

```diff
@@ -1904,4 +1970,45 @@ uct_component_t uct_ib_component = {
     .flags              = 0,
     .md_vfs_init        = uct_ib_md_vfs_init
 };
-UCT_COMPONENT_REGISTER(&uct_ib_component);
+UCT_COMPONENT_REGISTER_DEF(&uct_ib_component, ib);
+
+static void uct_ib_md_ops_add(uct_ib_md_ops_entry_t *entry)
+{
+    uct_ib_md_ops_entry_t *p;
+
+    ucs_list_for_each(p, &uct_ib_md_ops_list, list) {
+        if (p->priority < entry->priority) {
+            ucs_list_insert_before(&p->list, &entry->list);
+            return;
+        }
+    }
+
+    ucs_list_add_tail(&uct_ib_md_ops_list, &entry->list);
+}
+
+void UCS_F_CTOR uct_ib_init()
+{
+    int i;
+
+    uct_component_ib_ctor();
+
+    for (i = 0; i < ucs_static_array_size(uct_ib_tls); i++) {
+        ucs_list_add_tail(&ucs_config_global_list, &uct_ib_tls[i]->config.list);
+        ucs_list_add_tail(&uct_ib_component.tl_list, &uct_ib_tls[i]->list);
```

Comment:
can we add helper function (in UCT base) to unite with UCT_TL_REGISTER_DEF macro body?
e.g 
```
uct_tl_add(&component, &tl) {
  ucs_list_add_tail(&ucs_config_global_list, &tl->config.list)
  ucs_list_add_tail(&component->tl_list, &tl->list);
}
```